evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -0,0 +1,180 @@
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from evalml.pipelines import TimeSeriesRegressionPipeline
+
+
+@pytest.fixture
+def ts_data():
+    X, y = pd.DataFrame({"features": range(101, 132)}), pd.Series(range(1, 32))
+    y.index = pd.date_range("2020-10-01", "2020-10-31")
+    X.index = pd.date_range("2020-10-01", "2020-10-31")
+    return X, y
+
+
+@pytest.mark.parametrize("pipeline_class", [TimeSeriesRegressionPipeline])
+@pytest.mark.parametrize("components", [["One Hot Encoder"],
+                                        ["Delayed Feature Transformer", "One Hot Encoder"]])
+def test_time_series_pipeline_init(pipeline_class, components):
+
+    class Pipeline(pipeline_class):
+        component_graph = components + ["Random Forest Regressor"]
+
+    if "Delayed Feature Transformer" not in components:
+        pl = Pipeline({'pipeline': {"gap": 3, "max_delay": 5}})
+        assert "Delayed Feature Transformer" not in pl.parameters
+        assert pl.parameters['pipeline'] == {"gap": 3, "max_delay": 5}
+    else:
+        parameters = {"Delayed Feature Transformer": {"gap": 3, "max_delay": 5},
+                      "pipeline": {"gap": 3, "max_delay": 5}}
+        pl = Pipeline(parameters)
+        assert pl.parameters['Delayed Feature Transformer'] == {"gap": 3, "max_delay": 5,
+                                                                "delay_features": True, "delay_target": True}
+        assert pl.parameters['pipeline'] == {"gap": 3, "max_delay": 5}
+
+    assert Pipeline(parameters=pl.parameters) == pl
+
+    with pytest.raises(ValueError, match="gap and max_delay parameters cannot be omitted from the parameters dict"):
+        Pipeline({})
+
+
+@pytest.mark.parametrize("only_use_y", [True, False])
+@pytest.mark.parametrize("include_delayed_features", [True, False])
+@pytest.mark.parametrize("gap,max_delay", [(1, 2), (2, 2), (7, 3), (2, 4)])
+@pytest.mark.parametrize("pipeline_class,estimator_name", [(TimeSeriesRegressionPipeline, "Random Forest Regressor")])
+@patch("evalml.pipelines.components.RandomForestRegressor.fit")
+def test_fit_drop_nans_before_estimator(mock_regressor_fit, pipeline_class,
+                                        estimator_name, gap, max_delay, include_delayed_features, only_use_y, ts_data):
+
+    if only_use_y and not include_delayed_features:
+        pytest.skip("Can't prevent label leakage when only the target is used without lagging.")
```

Comment:
So this would happen when `max_delay=0` and `delay_target=True`? Perhaps we should raise an error in that case?

Reply:
Yea the problem is that all of our estimators require the X to be non-empty. If you are passing an X=None to the pipeline and you are not computing any features with a DelayedFeaturesTransformer, then our estimators will freak out. I'll raise an error in fit if we're in that case but we may need to rework this in the future when we add time-series specific estimators that compute their own features internally. In that case, we'll likely want to pass in X=None and not include a DelayFeatureTransformer in the pipeline.

Reply:
That's a good point @freddyaboulton . Sounds good, raising an error in pipeline fit makes sense, and then when we add timeseries support to `make_pipelines` for automl we'll just have to handle this edge case and not add delayed feature eng to the pipeline.